src/main/java/in/org/projecteka/hiu/dataprocessor/HealthDataProcessor.java

```diff
@@ -60,29 +65,46 @@ private void processEntries(DataContext context) {
         DataFlowRequestKeyMaterial keyMaterial = dataFlowRepository.getKeys(context.getTransactionId()).block();
         List<String> dataErrors = new ArrayList<>();
         context.getNotifiedData().getEntries().forEach(entry -> {
+            ProcessedResource processedResource = new ProcessedResource();
             if (hasContent(entry)) {
-                ProcessedResource processedResource = processEntryContent(context, entry, keyMaterial);
-                if (!processedResource.hasErrors()) {
-                    String resource =
-                            getEntryParser(entry.getMedia()).encodeResourceToString(processedResource.getResource());
-                    healthDataRepository.insertHealthData(
-                            context.getTransactionId(),
-                            context.getDataPartNumber(),
-                            resource,
-                            EntryStatus.SUCCEEDED)
-                            .block();
-                } else {
-                    dataErrors.addAll(processedResource.getErrors());
-                    healthDataRepository.insertHealthData(
-                            context.getTransactionId(),
-                            context.getDataPartNumber(),
-                            "",
-                            EntryStatus.ERRORED)
-                            .block();
+                processedResource = processEntryContent(context, entry, keyMaterial);
+            } else {
+                String url = String.format("%s?transactionId=%s", entry.getLink(), context.getTransactionId());
```

Comment:
It should not be done here even transactionId is needed, it should be part of link itself.

Reply:
I will add it as part of the link in HIP